src/Pelagos/Util/DOIutil.php

```diff
@@ -85,50 +88,23 @@ public function mintDOI(
         $status = 'reserved',
         $resourcetype = 'Dataset'
     ) {
-        $input = '_target:' . $this->escapeSpecialCharacters($url) . "\n";
-        $input .= "_profile:datacite\n";
-        $input .= "_status:$status\n";
-        $input .= 'datacite.creator:'
-            . $this->escapeSpecialCharacters($creator)
-            . "\n";
-        $input .= 'datacite.title:'
-            . $this->escapeSpecialCharacters($title)
-            . "\n";
-        $input .= 'datacite.publisher:' . $this->escapeSpecialCharacters($publisher) . "\n";
-        $input .= "datacite.publicationyear:$publicationYear\n";
-        $input .= "datacite.resourcetype:$resourcetype";
-
-        utf8_encode($input);
-
-        $ch = curl_init();
-        curl_setopt($ch, CURLOPT_URL, $this->url . '/shoulder/' . $this->doishoulder);
-        curl_setopt($ch, CURLOPT_USERPWD, $this->doiusername . ':' . $this->doipassword);
-        curl_setopt($ch, CURLOPT_POST, true);
-        curl_setopt(
-            $ch,
-            CURLOPT_HTTPHEADER,
-            array('Content-Type: text/plain; charset=UTF-8','Content-Length: ' . strlen($input))
-        );
-        curl_setopt($ch, CURLOPT_POSTFIELDS, $input);
-        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
-        $output = curl_exec($ch);
-        $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
-        curl_close($ch);
-
-        //check to see if it worked.
-        //using in array because EZID API returns 201 and EZ Datacite API returns 200.
-        if (!in_array($httpCode, [200, 201])) {
-            $expMsg = "ezid failed with:$httpCode($output)";
-            if ($httpCode >= 400 and $httpCode <= 499) {
-                throw new HttpClientErrorException($expMsg, $httpCode);
-            } elseif ($httpCode >= 500 or $httpCode == 0) {
-                throw new HttpServerErrorException($expMsg, $httpCode);
-            }
+        try {
+            $doi = $this->generateDoi();
+        } catch (HttpClientErrorException $exception) {
```

Comment:
So I looked around a bit, and rethrowing the same exception is not pretty.
For it to have meaning a different exception should be thrown at the right level.
i.e. DuplicateDoiException, HavingTroubleWithDataCiteException, CouldntCreateDoiException (examples).

Also, @praneethpr brought up that you can simplify this if you just put the doi (id) creation into the createDoi function, but leave the doi value as an option. (at the end or so).